static/scripts/dataprivacy/dataprivacy.js

```diff
@@ -1,255 +1,276 @@
-        updateButton(index);
-    }
-    function findLatestInvalid(to){
-        let i = 1;
-        for (; i <= to; i++) {
-            if(!isSectionValid(i)){
-                return i;
+            return this;
+        };
+    }
+    if(!NodeList.prototype.indexOf) {
+        NodeList.prototype.indexOf = function(node) {
+            return Array.from(this).indexOf(node);
+        };
+    }
+    if(!NodeList.prototype.filter) {
+        NodeList.prototype.filter = function(fct) {
+            return Array.from(this).filter(fct);
+        };
+    }
+    if(!NodeList.prototype.some) {
+        NodeList.prototype.some = function(fct) {
+            return Array.from(this).some(fct);
+        };
+    }
+    
+    /* MULTIPAGE INPUT FORM */
+    
+    function getMaxSelectionIndex() {
+        return document.querySelector(".stages").childElementCount;
+    }
+    
+    function getSelectionIndex() {
+        var radioButtons = document.querySelectorAll('.form input[type=radio]');
+        return radioButtons.indexOf(radioButtons.filter((node) => {
+            return node.checked;
+        })[0]) + 1;
+    }
+    
+    function showInvalid(sectionNr) {
+        document.querySelector(`section[data-panel="section-${sectionNr}"]`).classList.add("showInvalid");
+        document.querySelector(".content-wrapper").scrollTo(0, 0);
+    }
+    
+    function getSubmitPageIndex() {
+        return document.querySelectorAll('form .panels section').indexOf(document.querySelector(`section.submit-page`)) + 1
+    }
+    
+    function isSubmitted() {
+        return document.querySelector(".form").classList.contains('form-submitted');
+    }
+    
+    function setSelectionByIndex(index, event) {
+        event.preventDefault();
+        
+        function setSelection(index) {
+            document.querySelector('.form input[type="radio"]:nth-of-type(' + index + ')').checked = true;
+            updateButton(index);
+        }
+        
+        function findLatestInvalid(to) {
+            let i = 1;
+            for(; i <= to; i++) {
+                if(!isSectionValid(i)) {
+                    return i;
+                }
+            }
+            return to;
+        }
+        
+        const submitPageIndex = getSubmitPageIndex();
+        const submitted = isSubmitted();
+        if(submitted) {
+            if(index > submitPageIndex) {
+                setSelection(index)
+            } else { //prevent resubmit -> pages before unreachable
+                setSelection(submitPageIndex + 1)
+            }
+        } else {
+            index = Math.min(index, submitPageIndex); //prevent skip of submit
+            const latestInvalid = findLatestInvalid(index)
+            if(latestInvalid >= index) {
+                setSelection(index);
+            } else {
+                showInvalid(latestInvalid);
+                setSelection(latestInvalid);
+            }
+        }
+    }
+    
+    function updateButton(selectedIndex) {
```

Comment:
Das sieht meiner Meinung nach, nach einer Funktion aus, welche nur bei der registrierung notwendig ist. Ich würde sie daher nach `static/scripts/dataprivacy/registration.js` verschieben um den Code besser in anderen Formularen wiederverwenden zu können. In dieser Datei habe ich bisher nur Funktionen gehabt, welche allgemeine Funktionalität für Inputfelder bzw. Formulare mit mehreren Seiten bieten.